Source/WebCore/svg/SVGAnimationElement.cpp

```diff
@@ -442,7 +442,6 @@ float SVGAnimationElement::calculatePercentFromKeyPoints(float percent) const
     if (calcMode() == CalcMode::Discrete)
         return fromKeyPoint;
     
-    ASSERT(index + 1 < keyTimes.size());
```

Comment:
Why is this assertion removed? 

Looking at the logic in this function, I think it is only called when `keyTimes.size() > 1`. And `SVGAnimationElement::calculateKeyTimesIndex()` returns always value less than `keyTimes.size() - 1` if `keyTimes.size() > 1` and `calcMode() != CalcMode::Discrete`.

So this assertion should never fire.

Reply:
That was a mistake while backing out my prior changes.

Reply:
I have restored it, as well as fix the commit message to refer to the correct function for these new changes.